sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLEnv.scala

```diff
@@ -49,7 +49,10 @@ private[hive] object SparkSQLEnv extends Logging {
         .set(SQLConf.DATETIME_JAVA8API_ENABLED, true)
 
 
-      val sparkSession = SparkSession.builder.config(sparkConf).enableHiveSupport().getOrCreate()
+      val sparkSession = SparkSession.builder()
+        .config(sparkConf)
+        .config(HiveUtils.BUILTIN_HIVE_VERSION.key, HiveUtils.builtinHiveVersion)
```

Comment:
do we need to set it here? it's read-only now.

Reply:
This looks like another problem at the read-side that we need to look into. this PR just change it to read-only but still need to explicitly set the default value like before (https://github.com/apache/spark/pull/32200/files#diff-d6fa3cebb959651e721093f193f2c3b7197d7d0d1c503e34860aa048164b6b0eL66 and https://github.com/apache/spark/pull/32200/files#diff-4ade9276f3d22d5c47a42e1cd96dd6125e8ecc7d38e3f5f6841b8109936463b4L67). Otherwise, the existing and newly added tests would fail to get it.